File: src/main/java/com/naminhyeok/fantazzk/room/api/RoomResponse.java
package com.naminhyeok.fantazzk.room.api;

import com.naminhyeok.fantazzk.room.domain.Room;
import com.naminhyeok.fantazzk.room.domain.RoomStatus;
import java.util.List;

public record RoomResponse(
    String code,
    RoomStatus status,
    List<TeamLeaderResponse> teamLeaders
) {
    static RoomResponse from(Room room) {
        return new RoomResponse(
            room.getCode(),
            room.getStatus(),
            room.getLeaders().stream().map(TeamLeaderResponse::from).toList()
        );
    }
}

File: room/api/src/main/kotlin/com/naminhyeok/fantazzk/room/dto/RoomResponse.kt
package com.naminhyeok.fantazzk.room.dto

import com.naminhyeok.fantazzk.room.RoomModel
import com.naminhyeok.fantazzk.room.RoomStatus
import com.naminhyeok.fantazzk.room.RoomTeamLeaderModel

data class RoomResponse(
    val code: String,
    val status: RoomStatus,
    val teamLeaders: List<TeamLeaderResponse>,
) {
    companion object {
        fun from(
            room: RoomModel,
            leaders: List<RoomTeamLeaderModel>,
        ): RoomResponse =
            RoomResponse(
                code = room.code,
                status = room.status,
                teamLeaders = leaders.map { TeamLeaderResponse(it.teamLeaderId, it.nickname, it.remainingBudget) },
            )
    }
}
